packages/dashboard-core-plugins/src/panels/ConsolePanel.tsx

```diff
@@ -24,12 +34,62 @@ const DEFAULT_PANEL_STATE = Object.freeze({
   itemIds: [],
 });
 
-export class ConsolePanel extends PureComponent {
+interface ConsoleSettings {
+  isClosePanelsOnDisconnectEnabled?: boolean;
+}
+
+interface PanelState {
+  consoleSettings: ConsoleSettings;
+  itemIds: [string, string][];
+}
+
+type ItemIds = Map<string, string>;
+
+interface SessionWrapper {
+  session: IdeSession;
+  config: {
+    type: string;
+    id: string;
+  };
+}
+interface ConsolePanelProps {
+  commandHistoryStorage: CommandHistoryStorage;
+  glContainer: Container;
+  glEventHub: EventEmitter;
+
+  panelState: PanelState;
+
+  sessionWrapper: SessionWrapper;
+
+  timeZone: string;
+  unzip: (file: File) => Promise<JSZipObject[]>;
```

Comment:
`panelState` and `unzip` should be optional params:
```suggestion
  panelState?: PanelState;

  sessionWrapper: SessionWrapper;

  timeZone: string;
  unzip?: (file: File) => Promise<JSZipObject[]>;
```